File: Palindrome Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isListPalindrome(ListNode* &leftNode, ListNode* &rightNode)
    {
        bool result;
        if(rightNode == NULL)
          return true;
       
        bool resultSublist = isListPalindrome(leftNode, rightNode->next);
        
        if(!resultSublist) 
            return false;

        if(leftNode->val == rightNode->val)
            result = true; 
        else
            result = false;
        
        leftNode = leftNode->next;
        return result;
    }
    
    bool isPalindrome(ListNode* head) {
        ListNode *leftNode = head;
        ListNode *rightNode = head->next;
        return isListPalindrome(leftNode,rightNode);
    }
};

File: PalindromeLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isListPalindrome(self, leftNode, rightNode):
        if rightNode == None:
            return [True,leftNode]

        resultSublist = self.isListPalindrome(leftNode, rightNode.next)
        res = resultSublist[0]
        leftNode = resultSublist[1]

        if not res:
            return [False,None]

        if leftNode.val == rightNode.val:
            result = True
        else:
            result = False

        leftNode = leftNode.next

        return [result,leftNode]

    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        leftNode = head
        rightNode = head.next

        return self.isListPalindrome(leftNode,rightNode)[0]
